sql/table.cc

```diff
@@ -6450,6 +6478,38 @@ bool TABLE::check_read_removal(uint index)
   DBUG_RETURN(file->start_read_removal());
 }
 
+bool TABLE::has_compressed_columns_with_dictionaries() const
+{
+  DBUG_ENTER("has_compressed_columns_with_dictionaries");
+  DBUG_ASSERT(field != 0);
+
+  Field **field_ptr = field;
+  while(*field_ptr != 0 && !(*field_ptr)->has_associated_compression_dictionary())
+    ++field_ptr;
+
+  DBUG_RETURN(*field_ptr != 0);
+}
+
+void TABLE::update_compressed_columns_info(List<Create_field>& fields)
```

Comment:
Unfortunately, `List` and `List_iterator` have very questionable interfaces with regards to const-correctness. `List_iterator<T>` cannot be created with `const List<T>&`, for instance.

Method signature changed to `TABLE::update_compressed_columns_info(const List<Create_field>& fields)`. Used `const_cast<>` inside.